### PR TITLE
#17261 Fix and IT

### PR DESCRIPTION
**Pull request by nollymar**
Support plus sign on URLs

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/filters/CMSUrlUtil.java [hunk 1]
```diff
@@ -415,9 +415,18 @@ String xssCheck(String uri, String queryString) throws ServletException {
 	public String getURIFromRequest(HttpServletRequest request)
 			throws UnsupportedEncodingException {
 
-		return (request.getAttribute(CMS_FILTER_URI_OVERRIDE) != null) ? (String) request
+        String requestPath = request.getRequestURI();
+        try {
+            URI requestURI = new URI(requestPath);
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/filters/CMSUrlUtil.java]: Codacy Issue found: [Local variable 'requestURI' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4763071)

File: dotCMS/src/test/java/com/dotmarketing/filters/CMSUrlUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+     * Given Scenario: Call {@link CMSUrlUtil#getURIFromRequest(HttpServletRequest)}
+     * with a request that contains a mocked URI with a plus sign
+     * ExpectedResult: The URI returned must be the same one mocked,
+     * including the plus sign
+     */
+    @Test
+    public void testGetURIFromRequest() throws UnsupportedEncodingException {
+        final HttpServletRequest request = mock(HttpServletRequest.class);
+        when(request.getRequestURI()).thenReturn("dotcms+test.txt");
+        String result = CMSUrlUtil.getInstance().getURIFromRequest(request);
```
Comment (dev-dotcms) [inline dotCMS/src/test/java/com/dotmarketing/filters/CMSUrlUtilTest.java]: Codacy Issue found: [Local variable 'result' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4763071)

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/filters/CMSUrlUtil.java [hunk 3]
```diff
@@ -415,9 +415,18 @@ String xssCheck(String uri, String queryString) throws ServletException {
...
+            requestPath = requestURI.getPath();
+        } catch (URISyntaxException e) {
+            Logger.error(this, "Couldn't get URL from request " + requestPath, e);
+        }
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotmarketing/filters/CMSUrlUtil.java]: I think all this logic should go to a private method then we can call that method here

https://github.com/dotCMS/core/pull/17785/files#diff-b01fee3e8841e6997654dbe68e5baf2dR429

this way if "request.getAttribute(CMS_FILTER_URI_OVERRIDE) != null)" that code is not ran

File: dotCMS/src/test/java/com/dotmarketing/filters/CMSUrlUtilTest.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        assertEquals("dotcms+test.txt", result);
+    }
+
```
Comment (freddyucv) [inline dotCMS/src/test/java/com/dotmarketing/filters/CMSUrlUtilTest.java]: I know is out from the issue scope but I think is good idea create a test setting the CMS_FILTER_URI_OVERRIDE request attribute

**Review by freddyucv — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotmarketing/filters/CMSUrlUtil.java (hunk 3 above)
Comment (nollymar) [reply dotCMS/src/main/java/com/dotmarketing/filters/CMSUrlUtil.java]: Done

File: dotCMS/src/test/java/com/dotmarketing/filters/CMSUrlUtilTest.java (hunk 4 above)
Comment (nollymar) [reply dotCMS/src/test/java/com/dotmarketing/filters/CMSUrlUtilTest.java]: Done

File: dotCMS/src/test/java/com/dotmarketing/filters/CMSUrlUtilTest.java (hunk 2 above)
Comment (nollymar) [reply dotCMS/src/test/java/com/dotmarketing/filters/CMSUrlUtilTest.java]: Done

File: dotCMS/src/main/java/com/dotmarketing/filters/CMSUrlUtil.java (hunk 1 above)
Comment (nollymar) [reply dotCMS/src/main/java/com/dotmarketing/filters/CMSUrlUtil.java]: Done

**Review by freddyucv — APPROVED** [review]

**Pull request merged**